### ISSUE TITLE
Ability to pass ID to Box creation

### ISSUE THREAD

**Issue by gryznar** [objectbox/objectbox-dart#680]
### Is there an existing issue?

- [x] I have searched [existing issues](https://github.com/objectbox/objectbox-dart/issues)

### Use case

I have one entity used for dynamically created categories of data. Each of this category is separated from each other and then accessed in separate thread (isolate).

### Proposed solution

Allow passing custom optional id to `Box` object while creating it:
`final box = store.box<FooEntity>(id: 'foo')`

### Describe alternatives you've considered

1. One Box for all categories. It has few big dowsides:
- concurrent access via separate Isolates (or  threads in future Dart version) would not be efficient and performant
- size of data would be bigger (every entry would have to have additional parameter, assigning it to separate category)
- accessing data from single category unnecessary complicates the query (this additional param have to be included). This also slows down read, because whole data from different categories is kept in one place
2. Separate store for each category. This is performance overkill to have one store-one box setup
3. Create separate entity for each category. This is not possible, because amount of categories and categories themselves are dynamic

### Additional context

-

**Comment by greenrobot-team** [comment]
Thanks for reporting! ObjectBox does not really work well with a dynamic data model as its main purpose is to map a fixed data model to Dart classes.

Anyhow, option 1 is also what I would recommend on first thought. Obviously, there are up- and downsides which you seem to have analyzed pretty well (though I'm not sure how valid the concurrent access issue is). Based on that you will have to make a decision if this data model or even ObjectBox works for your use case.

Note: I labeled this issue with "more info required" so it will auto-close in a few days if there are no follow-up comments.

**Comment by gryznar** [comment]
As far as I understand, implementing such feature does not work well with fixed-data design?

**Comment by greenrobot-team** [comment]
Yes, that's basically the summary of what I said and why it is unlikely ObjectBox will ever support this.
